src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -951,6 +951,25 @@ const WorkflowView = _.flow(
         _.update('outputs', this.isSingle() ? () => ({}) : _.mapValues(_.trim))
       )
 
+      // Dockstore users who target floating tags can change their WDL via Github without explicitly selecting a new version in Terra.
+      // If the change removes or renames inputs or outputs, the `modifiedConfig` retrieved from the DB will include stale keys that
+      // are hidden from the UI and cause the workflow to fail at launch with "Validation errors: Extra inputs".
+      //
+      // The IO set `modifiedInputsOutputs` is always current because we regenerate it from the WDL each time; use it to filter out
+      // any extraneous IO keys before saving.
+      //
+      // https://broadworkbench.atlassian.net/browse/WA-291
+      const validInputs = _.map('name', modifiedInputsOutputs['inputs'])
+      const validOutputs = _.map('name', modifiedInputsOutputs['outputs'])
+
+      modifiedConfig['inputs'] = Object.fromEntries(Object.entries(modifiedConfig['inputs']).filter(function(tuple) {
```

Comment:
Nit: I think we prefer to use lodash `filter` (and probably `entries`). So this could be:
```
_.filter(tuple => _.includes(tuple[0], validInputs), _.fromPairs(_.toPairs(modifiedConfig['inputs'])))
```
(Or something similar. This hasn't been fully verified for syntactic correctness. See https://lodash.com/docs/#fromPairs for syntax. Although we use the functional version, so most of the time the args are switched.)

Reply:
It looks like
```
_.filter(_.toPairs(modifiedConfig['inputs']), pair => {return _.includes(desiredInputs, pair[0])})
```
works so it seems there is some truth to the arg switching statement (for both `includes` and `filter`)